app/src/app/layout/toast/toast.service.ts

```diff
@@ -16,29 +16,107 @@
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
 
-import { Injectable, TemplateRef } from '@angular/core';
+import {Injectable} from '@angular/core';
 
 @Injectable({
   providedIn: 'root'
 })
+
 export class ToastService {
 
-  toasts: any[] = [];
-  errorToast: any[] = [];
+  public toastsright: any[] = [];
+  public toastsleft: any[] = [];
+  public toastsdownleft: any[] = [];
+  public toastsdownright: any[] = [];
+  public toastscenter: any[] = [];
+
 
-  public show(textOrTpl: string | TemplateRef<any>, options: any = {}) {
-    this.toasts.push({ textOrTpl, ...options });
+  public showToast(text: string, options: any = {}, direction?: string) {
+    if (!options.icon) {
+      options.icon = 'fas fa-info-circle';
+    }
+    switch (direction) {
+      case 'right': {
+        this.toastsright.push({text, ...options});
+        break;
+      }
+      case 'left': {
+        this.toastsleft.push({text, ...options});
+        break;
+      }
+      case 'downleft': {
+        this.toastsdownleft.push({text, ...options});
+        break;
+      }
+      case 'downright': {
+        this.toastsdownright.push({text, ...options});
+        break;
+      }
+      case 'center': {
+        this.toastscenter.push({text, ...options});
+        break;
+      }
+      default: {
+        this.toastsright.push({text, ...options});
+        break;
+      }
+    }
   }
 
-  public error(textOrTpl: string | TemplateRef<any>, options: any = {}) {
-    this.errorToast.push({ textOrTpl, ...options });
+  public error(text: string, options: any = {}, direction?: string) {
```

Comment:
These are still no default parameters.
Look again [here](https://www.logicbig.com/tutorials/misc/typescript/function-optional-and-default-params.html) see exactly how this works